Pod/Extensions/SSO/SKYAuthContainer+SSO.m

```diff
@@ -0,0 +1,200 @@
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#import "SKYAuthContainer+SSO.h"
+#import "SKYAuthContainer_Private.h"
+#import "SKYWebOAuth.h"
+
+typedef enum : NSInteger { SKYOAuthActionLogin, SKYOAuthActionLink } SKYOAuthActionType;
+
+@implementation SKYAuthContainer (SSO)
+
+- (void)loginOAuthProvider:(NSString *)providerID
+                   options:(NSDictionary *)options
+         completionHandler:(SKYContainerUserOperationActionCompletion)completionHandler
+{
+    __weak typeof(self) weakSelf = self;
+    [self _oauthFlowWithProvider:providerID
+                         options:options
+                          action:SKYOAuthActionLogin
+               completionHandler:^(NSDictionary *_Nullable result, NSError *_Nullable error) {
+                   [weakSelf _handleLoginOAuthResult:result
+                                               error:error
+                                   completionHandler:completionHandler];
+               }];
+}
+
+- (void)linkOAuthProvider:(NSString *)providerID
+                  options:(NSDictionary *)options
+        completionHandler:(void (^_Nullable)(NSError *_Nullable))completionHandler
+{
+    [self _oauthFlowWithProvider:providerID
+                         options:options
+                          action:SKYOAuthActionLink
+               completionHandler:^(NSDictionary *_Nullable result, NSError *_Nullable error) {
+                   completionHandler(error);
+               }];
+}
+
+- (void)loginOAuthProvider:(NSString *_Nonnull)providerID
+               accessToken:(NSString *_Nonnull)accessToken
+         completionHandler:(SKYContainerUserOperationActionCompletion _Nullable)completionHandler
+{
+    __weak typeof(self) weakSelf = self;
+    [self.container callLambda:[self _getAuthURLWithAction:SKYOAuthActionLogin]
+           dictionaryArguments:@{@"access_token" : accessToken}
+             completionHandler:^(NSDictionary *result, NSError *error) {
+                 [weakSelf _handleLoginOAuthResult:result
+                                             error:error
+                                 completionHandler:completionHandler];
+             }];
+}
+
+- (void)linkOAuthProvider:(NSString *_Nonnull)providerID
+              accessToken:(NSString *_Nonnull)accessToken
+        completionHandler:(void (^_Nullable)(NSError *_Nullable))completionHandler
+{
+    [self.container callLambda:[self _getAuthURLWithAction:SKYOAuthActionLink]
+           dictionaryArguments:@{@"access_token" : accessToken}
+             completionHandler:^(NSDictionary *result, NSError *error) {
+                 completionHandler(error);
+             }];
+}
+
+- (void)_oauthFlowWithProvider:(NSString *)providerID
```

Comment:
IMO, it is better to put helper methods into their own class. This is because this file is a category on `SKYAuthContainer` and these helper methods are also added to `SKYAuthContainer`, but they don’t need to be there. Separating these helper methods help maintaining a clear separation between `SKYAuthContainer` and SSO-related methods.